precompiles/authorization/ics20_approve.go

```diff
@@ -0,0 +1,264 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package authorization
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	authzkeeper "github.com/cosmos/cosmos-sdk/x/authz/keeper"
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	channelkeeper "github.com/cosmos/ibc-go/v7/modules/core/04-channel/keeper"
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	cmn "github.com/evmos/evmos/v14/precompiles/common"
+)
+
+// TransferMsg is the ICS20 transfer message type.
+var TransferMsg = sdk.MsgTypeURL(&transfertypes.MsgTransfer{})
+
+// Approve implements the ICS20 approve transactions.
+func Approve(
+	ctx sdk.Context,
+	authzKeeper authzkeeper.Keeper,
+	channelKeeper channelkeeper.Keeper,
+	precompileAddr, grantee, origin common.Address,
```

Comment:
same qustion

Reply:
The origin is always the one approving for a grantee, otherwise we would have to change the signature of `Approve` to take both `grantee` and `granter` which is risky because anyone can create approvals for anyone.